IdentityCore/src/cache/MSIDMacKeychainTokenCache.m

```diff
@@ -0,0 +1,466 @@
+                                                             error:(__unused NSError **)error {
+    [self createUnimplementedError:error];
+    return nil;
+}
+
+// Remove items with the given Metadata key from the macOS keychain cache.
+- (BOOL)removeItemsWithMetadataKey:(__unused MSIDCacheKey *)key
+                           context:(__unused id<MSIDRequestContext>)context
+                             error:(__unused NSError **)error {
+    [self createUnimplementedError:error];
+    return FALSE;
+}
+
+#pragma mark - Wipe Info
+
+// Saves information about the app which most-recently removed a token.
+- (BOOL)saveWipeInfoWithContext:(__unused id<MSIDRequestContext>)context error:(__unused NSError **)error {
+    [self createUnimplementedError:error];
+    return FALSE;
+}
+
+// Read information about the app which most-recently removed a token.
+- (NSDictionary *)wipeInfo:(__unused id<MSIDRequestContext>)context error:(__unused NSError **)error {
+    [self createUnimplementedError:error];
+    return nil;
+}
+
+#pragma mark - clear
+
+// A test-only method that deletes all items from the cache for the given context.
+- (BOOL)clearWithContext:(__unused id<MSIDRequestContext>)context error:(__unused NSError **)error {
+    [self createUnimplementedError:error];
+    return FALSE;
+}
+
+#pragma mark - Utilities
+
+// Get the basic/default keychain query dictionary for account items.
+- (NSMutableDictionary *)defaultAccountQuery:(__unused MSIDCacheKey *)key {
+    MSID_TRACE;
+    NSMutableDictionary *query = [NSMutableDictionary new];
+
+    // Note: this initial implementation follows the cache schema, relying only on the basic key
+    // ("<homeAccountId>-<environment>-<realm>"). Additinal macOS-specific extensions
+    // are deferred for now until they are required.
+    query[(id)kSecClass] = (id)kSecClassGenericPassword;
+    if (key.account) {
+        query[(id)kSecAttrAccount] = key.account; // "<homeAccountId>-<environment>-<realm>"
+    }
+
+    return query;
+}
+
+// Determine whether the account item matches the cache key.
+- (BOOL)accountItem:(NSDictionary *)itemDict matchesKey:(MSIDCacheKey *)key {
+    MSID_TRACE;
+
+    // The MSIDCacheKey "account" property for MSIDAccountCacheItems is "<homeAccountId>-<environment>-<realm>".
+    // For multiple-account operations, it might be set to just "<homeAccountId>-<environment>-" or "<homeAccountId>--".
+    // For comparisons, start by looking for a "--" in key.account. Since <homeAccountId> is required,
+    // the double-dash is either at the end ("<homeAccountId>--") or in the middle ("<homeAccountId>--<realm>").
+    // In both cases, everything to the left of the double-dash needs to appear at the beginning
+    // of the keychain item's kSecAttrAccount attribute. If something appears after the double-dash,
+    // it needs to appear at the end of the attribute. If the double-dash isn't present, use
+    // the full key.account string for matching.
```

Comment:
This is unnecessary if top level filtering by keys is used.

Reply:
This has been eliminated, following the iOS keychian implementation approach.